### ISSUE TITLE
lldpd fails to send PDUs on interfaces with MTU=1500, on systems with mixed MTU settings

### ISSUE THREAD

**Issue by PsichoD** [vincentbernat/lldpd#365]
# Bug description
We have a server with 4 interfaces. Two of them (em1,em2) are configured with MTU=9000 and the other two(p3p1,p3p2) are configured with MTU=1500. em1 and em2 can successfully send PDU(confirmed by tcpdump), but p3p1 and p3p2's PDUs are not being sent while LLDPD's debug out put shows them as 'sent'. 

## Steps to reproduce the problem

 1. Install lldpd rpm: `yum install lldpd`.
 
2. Run `/usr/sbin/lldpd -D main -D interfaces -D send -D receive -D loop -dddd`
  
3. Run `tcpdump -i p3p1 ether proto 0x88cc -vv` in a separate terminal.
 
## Expected outcome

Should trace `lldpd` PDUs sent by the server.

## Current outcome
Only tracing switch PDUs:

16:35:30 root@nyc01node31:~# tcpdump -i p3p1 ether proto 0x88cc -vv
tcpdump: WARNING: p3p1: no IPv4 address assigned
tcpdump: listening on p3p1, link-type EN10MB (Ethernet), capture size 65535 bytes
16:35:44.669072 LLDP, length 190

# Additional information

 - Output of `lldpd -vv`:
 
```
lldpd 1.0.4
  Built on 2019-08-13T12:02:30Z

Additional LLDP features:    LLDP-MED, Dot1, Dot3, Custom TLV
Additional protocols:        CDP, FDP, EDP, SONMP
SNMP support:                yes
Old kernel support:          no (Linux 2.6.39+)
Privilege separation:        enabled
Privilege separation user:   lldpd
Privilege separation group:  lldpd
Privilege separation chroot: /run/lldpd/chroot
Configuration directory:     /etc

C compiler command: gcc -std=gnu99  -fdiagnostics-show-option -fdiagnostics-color=auto -pipe -Wall -W -Wextra -Wformat -Wformat-security -Wfatal-errors -Wcast-align -Winline -Wpointer-arith -fno-omit-frame-pointer -Wno-unused-parameter -Wno-missing-field-initializers -Wno-sign-compare -fstack-protector -fstack-protector-all -fstack-protector-strong -fstack-clash-protection  -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mtune=generic 
Linker command:     /usr/bin/ld -m elf_x86_64  -Wl,-z,relro -Wl,-z,now  -Wl,-z,relro
```

 - Output of `ps -fp $(pgrep -d, -x lldpd)`:
 
```
UID        PID  PPID  C STIME TTY          TIME CMD
root     41439     1  0 16:19 ?        00:00:00 /usr/sbin/lldpd
lldpd    41441 41439  0 16:19 ?        00:00:00 /usr/sbin/lldpd

```

 - Output of `uname -sro`:
 
```
Linux 3.10.0-862.11.6.el7.x86_64 GNU/Linux
```

 - Output of `tcpdump -pni eth0 -vv -X ether host 01:80:c2:00:00:0e`:
 
```
16:35:44.669072 LLDP, length 190
	Chassis ID TLV (1), length 7
	  Subtype MAC address (4): 00:1c:73:c6:8a:ab (oui Unknown)
	  0x0000:  0400 1c73 c68a ab
[...]
       0x0000:  4172 6973 7461 204e 6574 776f 726b 7320
	  0x0010:  454f 5320 7665 7273 696f 6e20 342e 3134
	  0x0020:  2e31 314d 2072 756e 6e69 6e67 206f 6e20
	  0x0030:  616e 2041 7269 7374 6120 4e65 7477 6f72
	  0x0040:  6b73 2044 4353 2d37 3135 3053 2d36 342d
	  0x0050:  434c
```

**Comment by vincentbernat** [comment]
What kind of interfaces are `p3p1` and `p3p2`? What's the output of `ethtool -i p3p1`?

**Comment by PsichoD** [comment]
Those are 'Solarflare Flareon Ultra 7000 Series 10G Adapter' cards:

17:12:33 root@nyc01node31:~# ethtool -i p3p1
driver: sfc
version: 4.13.1.1034
firmware-version: 4.7.1.1001 rx1 tx1
expansion-rom-version:
bus-info: 0000:04:00.0
supports-statistics: yes
supports-test: yes
supports-eeprom-access: no
supports-register-dump: yes
supports-priv-flags: yes
17:12:39 root@nyc01node31:~# ethtool -i p3p2
driver: sfc
version: 4.13.1.1034
firmware-version: 4.7.1.1001 rx1 tx1
expansion-rom-version:
bus-info: 0000:04:00.1
supports-statistics: yes
supports-test: yes
supports-eeprom-access: no
supports-register-dump: yes
supports-priv-flags: yes
17:12:42 root@nyc01node31:~#

BTW, when running 'lldpd -I p3p1,p3p2' the PDUs are sent correctly.

**Comment by vincentbernat** [comment]
❦ 19 novembre 2019 14:15 -08, PsichoD <notifications@github.com>:

> BTW, when running 'lldpd -I p3p1,p3p2' the PDUs are sent correctly.

Oh, OK, I'll investigate a bit. It should be easy to reproduce.

**Comment by vincentbernat** [comment]
I am not able to reproduce. Could you run `lldpd` with `-D send -ddd` and provide the output?

**Comment by PsichoD** [comment]
Hi Vincent,

Apologies for the late reply. Here is the output of 'lldpd with -D send
-dddd' and the packet capture for the same time period:

03:16:31 root@localhost:~# lldpd with -D send -dddd
2019-11-21T03:17:21 [INFO/privsep] no libcap support, running monitor as
root
2019-11-21T03:17:21 [INFO/main] protocol LLDP enabled
2019-11-21T03:17:21 [INFO/main] protocol CDPv1 disabled
2019-11-21T03:17:21 [INFO/main] protocol CDPv2 disabled
2019-11-21T03:17:21 [INFO/main] protocol SONMP disabled
2019-11-21T03:17:21 [INFO/main] protocol EDP disabled
2019-11-21T03:17:21 [INFO/main] protocol FDP disabled
2019-11-21T03:17:21 [INFO/event] libevent 2.0.21-stable initialized with
epoll method
2019-11-21T03:17:21 [ DBG/lldpctl] cannot find configuration file/directory
/etc/lldpd.conf
2019-11-21T03:17:21 [ DBG/lldpctl] input: /etc/lldpd.d/README.conf
2019-11-21T03:17:21 [ DBG/lldpctl] connect to lldpd
2019-11-21T03:17:21 [ DBG/lldpctl] process: /etc/lldpd.d/README.conf
2019-11-21T03:17:21 [ DBG/lldpctl] tokenize command line
2019-11-21T03:17:21 [ DBG/lldpctl] tokenize command line
2019-11-21T03:17:21 [ DBG/lldpctl] tokenize command line
2019-11-21T03:17:21 [ DBG/lldpctl] tokenize command line
2019-11-21T03:17:21 [ DBG/lldpctl] tokenize command line
2019-11-21T03:17:21 [ DBG/lldpctl] tokenize command line
2019-11-21T03:17:21 [ DBG/lldpctl] tokenize command line
2019-11-21T03:17:21 [ DBG/lldpctl] tokenize command line
2019-11-21T03:17:21 [ DBG/lldpctl] argument 00: `resume`
2019-11-21T03:17:21 [ DBG/lldpctl] process argument 00: `resume`
2019-11-21T03:17:21 [ DBG/lldpctl] process argument 01: `<CR>`
2019-11-21T03:17:21 [ DBG/control] send a message through control socket
2019-11-21T03:17:21 [ DBG/control] connect to control socket
/run/lldpd/lldpd.socket
2019-11-21T03:17:21 [WARN/interfaces] cannot get ethtool link information
with GLINKSETTINGS (requires 4.9+). 25G+ speeds may be missing in MAC/PHY
TLVs
2019-11-21T03:17:21 [ DBG/control] receive a message through control socket
2019-11-21T03:17:21 [ DBG/control] receive a message through control socket
2019-11-21T03:17:21 [ DBG/control] receive a message through control socket
2019-11-21T03:17:21 [ DBG/control] receive a message through control socket
2019-11-21T03:17:21 [ DBG/marshal] start unserialization of lldpd_config
2019-11-21T03:17:21 [ DBG/marshal] start unserialization of null string
2019-11-21T03:17:21 [ DBG/control] send a message through control socket
2019-11-21T03:17:21 [ DBG/marshal] start serialization of lldpd_config
2019-11-21T03:17:21 [ DBG/marshal] start serialization of null string
2019-11-21T03:17:21 [ DBG/send] send PDU on em1
2019-11-21T03:17:21 [ DBG/send] fallback to protocol LLDP for em1
2019-11-21T03:17:21 [ DBG/send] send PDU on em2
2019-11-21T03:17:21 [ DBG/send] fallback to protocol LLDP for em2
2019-11-21T03:17:21 [ DBG/send] send PDU on p3p1
2019-11-21T03:17:21 [ DBG/send] fallback to protocol LLDP for p3p1
2019-11-21T03:17:21 [ DBG/send] send PDU on p3p2
2019-11-21T03:17:21 [ DBG/send] fallback to protocol LLDP for p3p2
2019-11-21T03:17:21 [ DBG/send] send PDU on idrac
2019-11-21T03:17:21 [ DBG/send] fallback to protocol LLDP for idrac
2019-11-21T03:17:21 [ DBG/control] receive a message through control socket
2019-11-21T03:17:21 [ DBG/control] receive a message through control socket
2019-11-21T03:17:21 [INFO/lldpctl] lldpd should resume operations
2019-11-21T03:17:21 [ DBG/alloc] general configuration cleanup
2019-11-21T03:17:51 [ DBG/send] send PDU on p3p2
2019-11-21T03:17:51 [ DBG/send] send PDU on p3p2 with protocol LLDP
2019-11-21T03:17:51 [ DBG/send] send PDU on p3p1
2019-11-21T03:17:51 [ DBG/send] send PDU on p3p1 with protocol LLDP
2019-11-21T03:17:51 [ DBG/send] send PDU on em2
2019-11-21T03:17:51 [ DBG/send] send PDU on em2 with protocol LLDP
2019-11-21T03:17:51 [ DBG/send] send PDU on em1
2019-11-21T03:17:51 [ DBG/send] send PDU on em1 with protocol LLDP
2019-11-21T03:17:51 [ DBG/send] send PDU on idrac
2019-11-21T03:17:51 [ DBG/send] fallback to protocol LLDP for idrac
03:17:53 root@localhost:~#

**Comment by vincentbernat** [comment]
Can you add `-D interfaces` and check if you get `send PDU to ethernet device p3p1 (fd=XX)`?
Would you be able to change the MTU of em1 and em2 to 1500 and check if it really solves your issue?

**Comment by PsichoD** [comment]
Hi Vincent,

I found the issue with our config and I also failed to mention an important
detail: p3p1 and p3p2 are slave interfaces to an LACP bond0, which in turn
has many VLANs configured on as bond1.123, bond1.124 and so forth.
bond0 was configured with MTU=1500, while some of the VLAN interfaces were
configured with MTU=9000.
Once I changed bond0 to use MTU=9000, lldpd on p3p1 and p3p2 started
working.
bond0 forced the slave interfaces p3p1 and p3p2 to MTU=9000 as well without
me having to change the config.

Thanks for helping out!
